example/progress.js

```diff
@@ -2,16 +2,25 @@ const path = require('path');
 const fs   = require('fs');
 const ytdl = require('..');
 
-var url = 'https://www.youtube.com/watch?v=WhXefyLs-uw';
-var output = path.resolve(__dirname, 'video.mp4');
+const url = 'https://www.youtube.com/watch?v=WhXefyLs-uw';
+const output = path.resolve(__dirname, 'video.mp4');
 
-var video = ytdl(url);
+const video = ytdl(url);
+let starttime;
 video.pipe(fs.createWriteStream(output));
-video.on('progress', function(chunkLength, downloaded, total) {
+video.once('response', () => {
+  starttime = Date.now();
+})
+video.on('progress', (chunkLength, downloaded, total) => {
+  const float_downloaded = downloaded / total;
+  const downloaded_min = (Date.now() - starttime) / 1000 / 60;
   process.stdout.cursorTo(0);
   process.stdout.clearLine(1);
-  process.stdout.write((downloaded / total * 100).toFixed(2) + '% ');
+  process.stdout.write(`${(float_downloaded * 100).toFixed(2)}% downloaded`);
+  process.stdout.write(`(${downloaded / 1024 / 1025}MB of ${total / 1024 / 1024}MB)`);
+  process.stdout.write(`, running for: ${download_time.toFixed(2)}minutes`);
+  process.stdout.write(`, estimated time left: ${(download_time / float_downloaded - download_time).toFixed(2)}minutes`);
```

Comment:
thought about extra lines but since i didn't know how to clear them again i just stuffed all in one
works fine thou and i guess making it pretty would be 2 much effort